src/libraries/Common/src/Interop/Unix/System.Native/Interop.PosixSignal.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_SetPosixSignalHandler")]
+        [SuppressGCTransition]
+        internal static extern unsafe void SetPosixSignalHandler(delegate* unmanaged<int, PosixSignal, int> handler);
+
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_EnablePosixSignalHandling")]
+        [SuppressGCTransition]
+        internal static extern void EnablePosixSignalHandling(int signal);
+
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_DisablePosixSignalHandling")]
```

Comment:
`SystemNative_DisablePosixSignalHandling` takes a lock. Not safe for SuppressGCTransition.